src/clr-angular/data/datagrid/datagrid-row.html

```diff
@@ -3,10 +3,17 @@
   Clicking of that wrapper label will equate to clicking on the whole row, which triggers the checkbox to toggle.
 -->
 <label class="datagrid-row-clickable" *ngIf="selection.rowSelectionMode">
-  <ng-template [ngTemplateOutlet]="rowContent"></ng-template>
+  <clr-expandable [clrExpandTrigger]="expanded" *ngIf="globalExpandable.hasExpandableRow">
```

Comment:
Interesting. So placing these animations one level deep inside here instead of the host element helping us to avoid `setTimeout`. I think the animation might have been too eager because it didn't wait for all these templates to be inserted/outlet first. So we should probably try to placing the animation on the div that's wrapping the row content. That might solve this whole animation issue without creating an extra wrapper component.

Reply:
The main problem is that there is a very specific time slot when the animation has to start, that's between rendering the content and updating the parent that wraps it.
If it's too early - the animation finishes before we have the content loaded.
If it's too late - then the wrapper gets expanded before the animation takes place and in the best case this leads to flicker, like this:
1) initial->100%, immediate on wrapper fit to content
2) back to initial->100%, as requested by animation
Step 1) may seem questionable and it can be skipped if we keep static initial height, but this will make it impossible to auto resize the content after the animation, for example with window resize, DG column resize, etc. Which we do not want.

The initial idea behind the component was to take advantage of the onChanges lifecycle hook of the component and/or try to add auto-detection of content size changes (without using the trigger), but this has so far been unsuccessful and greatly increases complexity of the solution. 
I find having a separate component for this even in the current state useful, because:
1) It separates the animation logic from the component.
2) Makes it reusable.
3) If we decide to improve on this later, we have to do it on one place only.

Reply:
If you still find the component for-animation solution the most viable one, I am willing to go forward with it. Regarding the reusability of the animations, we can have it in exportable functions just like we currently do in `utils/animations` which is what is recommended in Angular doc. But my real reluctance to go with a component for animation was that it could create just another layer of complexity related to change detection. That's why I recommended trying out the animations on the exact element that expands/collapse. Another thing we should note is that if use this current solution inside a component that queries its children components with ContentChildren/ContentChild, we would break their direct parent/child relationship by having an extra wrapper between them.

Reply:
The animation component is part of the ViewChildren of the ClrDatagridRow component. It's not supplied by the user, but used internally in the component. Whether it's a div or a clr-expandable-animation, it's all the same - no effect on the parent/child relationship with the cells which are ContentChildren.

Reply:
And second, about exporting it as a function, I don't think the animation definition on its own is some state of the art technology, the tricky parts are in supplying and evaluating the startHeight parameter.
Also, you see that this issue alone already takes about a month to set it all up properly. I think the benefit would be in having a component where the fine details are glued together, and only a minor set of inputs are left for wiring it into a new component.